copy_android_resources.js

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env node
+module.exports = copyAndroidResources;
+
+var source = "resources/android/custom/";
+var android_dests = [
+        "res/drawable-ldpi-v4/",
+        "res/drawable-mdpi-v4/",
+        "res/drawable-hdpi-v4/",
+        "res/drawable-xhdpi-v4/",
+        "res/drawable-xxhdpi-v4/",
+        "res/drawable-xxxhdpi-v4/",
+        "res/drawable-ldpi/",
+        "res/drawable-mdpi/",
+        "res/drawable-hdpi/",
+        "res/drawable-xhdpi/",
+        "res/drawable-xxhdpi/",
+        "res/drawable-xxxhdpi/"
+        ];
+
+
+function copyAndroidResources(ctx){
+    if(ctx.opts.platforms.indexOf('android') < 0){
+        return;
+    }
+    var fs       = ctx.requireCordovaModule('fs'),
+        path     = ctx.requireCordovaModule('path'),
+        Q = ctx.requireCordovaModule('q');
+    
+    var android_rootdir = path.join(ctx.opts.projectRoot, 'platforms/android');
+
+    function copy(src, dest_dir, file){
+        var deferred = Q.defer();
+        if (fs.existsSync(src) && fs.existsSync(dest_dir)) {
+            deferred.resolve(fs.createReadStream(src).pipe(
+               fs.createWriteStream(path.join(dest_dir, file))));
+        }
+        else{
+            deferred.resolve();
+        }
+        return deferred.promise;
+    }
+
+    var promise = Q(-1);
```

Comment:
Doing this with the Q library just immediately resolves the promise with the value `-1`. This means cordova immediately thinks this hook is completed, which could result in weird race conditions or other errors.